test/katex-spec.js

```diff
@@ -2451,33 +2452,69 @@ describe("operatorname support", function() {
     });
 });
 
-describe("An href command", function() {
+describe("href and url commands", function() {
+    // We can't use raw strings for \url because \u is for Unicode escapes.
+
     it("should parse its input", function() {
-        expect`\href{http://example.com/}{example here}`.toParse();
+        expect`\href{http://example.com/}{example here}`.toBuild();
+        expect("\\url{http://example.com/}").toBuild();
+    });
+
+    it("should allow empty URLs", function() {
+        expect`\href{}{example here}`.toBuild();
+        expect("\\url{}").toBuild();
+    });
+
+    it("should allow single-character URLs", () => {
+        expect`\href%end`.toParseLike("\\href{%}end");
+        expect`\href %end`.toParseLike("\\href{%}end");
+        expect("\\url%end").toParseLike("\\url{%}end");
+        expect("\\url %end").toParseLike("\\url{%}end");
```

Comment:
Why the mix of 
```
expect`...`
```
and 
```
expect(...)
```
?

Reply:
`\u` doesn't work in a raw string, because ES thinks it's the beginning of a Unicode escape. (This is mentioned in our documentation.) This means we need to use the old fashioned method for `\url`, sadly. (`\\u` also doesn't work.)

Reply:
There was a revision to the template string spec for ES2018.  I assumed that babel supported it but after checking http://kangax.github.io/compat-table/es2016plus/ it appears not.